From: Thomas Guyot-Sionnest <dermoth@aei.ca>
Date: Wed, 20 May 2009 04:20:11 +0000 (-0400)
Subject: Fix check_mysql* not using password set in my.cnf (#2531905 - Ben Timby)
X-Git-Url: http://repo.or.cz/w/nagiosplugins.git?a=commitdiff_plain;h=917fcc7c302f67b42482a1a3e16e62f3b39d4e80

Fix check_mysql* not using password set in my.cnf (#2531905 - Ben Timby)
---

diff --git a/plugins/check_mysql.c b/plugins/check_mysql.c
index 689648e..05d2b01 100644
--- a/plugins/check_mysql.c
+++ b/plugins/check_mysql.c
@@ -5,7 +5,7 @@
 * License: GPL
 * Copyright (c) 1999 Didi Rieder (adrieder@sbox.tu-graz.ac.at)
 * Copyright (c) 2000 Karl DeBisschop (kdebisschop@users.sourceforge.net)
-* Copyright (c) 1999-2007 Nagios Plugins Development Team
+* Copyright (c) 1999-2009 Nagios Plugins Development Team
 * 
 * Description:
 * 
@@ -358,9 +358,6 @@ validate_arguments (void)
 	if (db_host == NULL)
 		db_host = strdup("");
 
-	if (db_pass == NULL)
-		db_pass = strdup("");
-
 	if (db == NULL)
 		db = strdup("");
 
@@ -417,6 +414,9 @@ print_help (void)
 	printf ("\n");
 	printf ("%s\n", _("Notes:"));
 	printf (_(UT_EXTRA_OPTS_NOTES));
+	printf ("\n");
+	printf (" %s\n", _("You must specify -p with an empty string to force an empty password,"));
+	printf (" %s\n", _("overriding any my.cnf settings."));
 #endif
 
 	printf (_(UT_SUPPORT));
diff --git a/plugins/check_mysql_query.c b/plugins/check_mysql_query.c
index 802e345..47dd861 100644
--- a/plugins/check_mysql_query.c
+++ b/plugins/check_mysql_query.c
@@ -3,7 +3,7 @@
 * Nagios check_mysql_query plugin
 * 
 * License: GPL
-* Copyright (c) 2006-2007 Nagios Plugins Development Team
+* Copyright (c) 2006-2009 Nagios Plugins Development Team
 * Original code from check_mysql, copyright 1999 Didi Rieder
 * 
 * Description:
@@ -266,9 +266,6 @@ validate_arguments (void)
 	if (db_host == NULL)
 		db_host = strdup("");
 
-	if (db_pass == NULL)
-		db_pass = strdup("");
-
 	if (db == NULL)
 		db = strdup("");
 
@@ -317,6 +314,9 @@ print_help (void)
 	printf ("\n");
 	printf ("%s\n", _("Notes:"));
 	printf (_(UT_EXTRA_OPTS_NOTES));
+	printf ("\n");
+	printf (" %s\n", _("You must specify -p with an empty string to force an empty password,"));
+	printf (" %s\n", _("overriding any my.cnf settings."));
 #endif
 
 	printf (_(UT_SUPPORT));
